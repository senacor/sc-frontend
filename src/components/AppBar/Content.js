import React from 'react';
import { withStyles } from '@material-ui/core';
import Message from './Message';

const drawerWidth = 270;
const styles = theme => ({
  ...theme,
  content: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    width: '100%',
    height: '100%',
    minHeight: '100vh',
    [theme.breakpoints.up('lg')]: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth
    }
  },
  toolbar: theme.mixins.toolbar
});

const Content = ({ classes, children }) => {
  return (
    <main className={classes.content}>
      <div className={classes.toolbar} />
      <Message />
      {children}
    </main>
  );
};

export default withStyles(styles)(Content);
