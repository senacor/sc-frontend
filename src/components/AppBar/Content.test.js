import React from 'react';
import { shallowWithIntl } from 'enzyme-react-intl';

import Content from './Content';

const mockChildren = 'child';

describe('Content', () => {
  const wrapper = shallowWithIntl(<Content children={mockChildren} />);
  it('checks if Content has children', () => {
    expect(wrapper.props().children).toBeTruthy();
  });
});
