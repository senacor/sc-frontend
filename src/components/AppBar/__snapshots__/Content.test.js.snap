// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render correctly 1`] = `
<Content
  classes={
    Object {
      "breakpoints": "Content-breakpoints-1",
      "content": "Content-content-13",
      "direction": "Content-direction-2",
      "mixins": "Content-mixins-3",
      "overrides": "Content-overrides-4",
      "palette": "Content-palette-5",
      "props": "Content-props-6",
      "shadows": "Content-shadows-7",
      "shape": "Content-shape-9",
      "spacing": "Content-spacing-10",
      "toolbar": "Content-toolbar-14",
      "transitions": "Content-transitions-11",
      "typography": "Content-typography-8",
      "zIndex": "Content-zIndex-12",
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  child
</Content>
`;
